WordPress/Classes/ViewRelated/Stats/Extensions/Double+Stats.swift

```diff
@@ -143,6 +143,14 @@ extension Double {
         return formattedString
     }
 
+    func abbreviatedPercentageString(forHeroNumber: Bool = false) -> String {
+        guard let numberWithPercentage = NumberFormatter.statsPercentage.string(from: .init(value: self)) else {
+            return abbreviatedString(forHeroNumber: forHeroNumber)
+        }
+
+        return numberWithPercentage.replacingOccurrences(of: "\(self)", with: self.abbreviatedString(forHeroNumber: forHeroNumber))
+    }
+
```

Comment:
How should large percentages such as 10000% be formatted? This code formats them as "10,000%" which I think is correct (an alternative might be e.g. "10.0K%"). 

I think this code formats correctly, but there's some code that's never called.

From what I can tell, the call to `replacingOccurrences(of: with:)` will not have any effect.

Since `statsPercentage` applies formatting:
```
(lldb) po NumberFormatter.statsPercentage.string(from: .init(value: 10000))
▿ Optional<String>
  - some : "10,000%"
```

There are no occurrences of the original unformatted string in `numberWithPercentage`:
```
(lldb) po numberWithPercentage.replacingOccurrences(of: "\(self)", with: self.abbreviatedString(forHeroNumber: forHeroNumber))
"10,000%"
```

For smaller percentages like 1000 the same applies:
```
(lldb) po NumberFormatter.statsPercentage.string(from: .init(value: 1000))
▿ Optional<String>
  - some : "1,000%"
(lldb) po "1,000%".replacingOccurrences(of: "\(1000)", with: self.abbreviatedString(forHeroNumber: false))
"1,000%"
```

For even smaller numbers, the search-replace has no effect because it replaces with the same thing:
```
(lldb) po "100%".replacingOccurrences(of: "\(100)", with: 100.abbreviatedString(forHeroNumber: false))
"100%"
```

This means that there will be no abbreviation (even though it's in the function's name), and `forHeroNumber` will not be used. So I think it makes sense to simplify the function like this:

```
   func percentageString() -> String {
        guard let numberWithPercentage = NumberFormatter.statsPercentage.string(from: .init(value: self)) else {
            return "\(self) %"
        }

        return numberWithPercentage
    }
```


What do you think?

Also, it should be safe to force-unwrap the `string(from:)` since [there doesn't seem to be any chance of it returning nil for doubles](https://stackoverflow.com/questions/48504597/why-does-numberformatters-stringfrom-return-an-optional), but happy to go with what you think is best here.

Reply:
What a great observation. Different locales have different grouping separators (and some don't), so this naive replacing strategy doesn't work in a lot of cases.

> an alternative might be e.g. "10.0K%

That was my intention, since previously `abbreviatedString()`% was used, so I wanted to keep the abbreviation logic and localize percentage placement. However, I don't think this abbreviation is necessary for percentages, they will rarely be that high.

> This means that there will be no abbreviation (even though it's in the function's name), and forHeroNumber will not be used. So I think it makes sense to simplify the function like this:

Yes, I agree. 

> Also, it should be safe to force-unwrap the string(from:) since [there doesn't seem to be any chance of it returning nil for doubles](https://stackoverflow.com/questions/48504597/why-does-numberformatters-stringfrom-return-an-optional), but happy to go with what you think is best here.

Yes, the arguments look convincing. The `nil` values look to be only possible if using some weird trickery.

Reply:
> and some don't

I wasn't aware that some locales don't have thousand separators, this is good to know 👍 

> since previously abbreviatedString()% was used

I see, that's a good point.